### DISCUSSION TITLE
feature: range of normal attack

### DISCUSSION THREAD

**Discussion by WeylonSantana** [AscensionGameDev/Intersect-Engine#1545] [Ideas]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when I want the normal attack to reach more than 1 tile, like 2 or 3 to simulate a short-medium range weapon, like long sword, spear and etc.

**Describe the solution you'd like**
In the area of ​​armed and unarmed combat, in addition to the damage, define how many tiles the attack reaches
One more numeric up down

**Additional context**

**Comment by Daywalkrr** [comment]
It would be cool to implement this in a similar way to, say, the projectile spawn editor - where we show the user a grid and allow them to pick the tiles that are and are not valid "hit tiles", as well as add a checkbox for whether or not we should avoid block tiles.

This could also be extended to NPCs but that is best left for a separate feature.

EDIT: A tough thing to consider: how should animations play?

**Comment by Arufonsu** [comment]
> This could also be extended to NPCs but that is best left for a separate feature.

Sounds like a good thing for the _NPC Script Editor_ idea that's been discussed here and there.

But for now, these aditional features should be added to close-combat weapons held by players only.
